tests/solana.rs

```diff
@@ -572,6 +572,63 @@ impl SyscallObject<UserError> for SolKeccak256 {
         *result = Ok(0)
     }
 }
+struct SolCreateProgramAddress();
+impl SolCreateProgramAddress {
+    /// new
+    pub fn init<C, E>(_unused: C) -> Box<dyn SyscallObject<UserError>> {
+        Box::new(Self {})
+    }
+}
+
+impl SyscallObject<UserError> for SolCreateProgramAddress {
+    fn call(
+        &mut self,
+        seed_ptr: u64,
+        seed_len: u64,
+        program_id: u64,
+        dest: u64,
+        _arg5: u64,
+        memory_mapping: &mut MemoryMapping,
+        result: &mut Result<u64, EbpfError<UserError>>,
+    ) {
+        assert!(seed_len <= 16);
+
+        let arrays = question_mark!(
+            translate_slice::<(u64, u64)>(memory_mapping, seed_ptr, seed_len),
+            result
+        );
+
+        let mut seeds = Vec::new();
+
+        for (addr, len) in arrays {
+            assert!(*len < 32);
+
+            let buf = question_mark!(translate_slice::<u8>(memory_mapping, *addr, *len), result);
```

Comment:
What is `question_mark`?

Reply:
It's a macro.

Reply:
It's imported from solana_rbpf

Reply:
You want to catch errors, so that the system call can return an error. But you don't want the function call itself to fail, otherwise the solana VM code would abort. So, this puts the result from the call into `result` and then system call returns, without returning an error (`result` is passed in).